Source/Core/Core/Debugger/PPCDebugInterface.cpp

```diff
@@ -163,6 +165,15 @@ void PPCDebugInterface::ToggleMemCheck(unsigned int address)
 void PPCDebugInterface::InsertBLR(unsigned int address, unsigned int value)
 {
   PowerPC::HostWrite_U32(value, address);
+  if (CPU::GetState() == CPU::State::CPU_RUNNING)
+  {
+    PowerPC::SchedulateInvalidateCacheThreadSafe(address);
+    CoreTiming::Advance();
```

Comment:
Don't call this from the GUI thread, all sorts of weird things will happen.

You don't need to call it at all.